firmware/src/led_ring/led_ring_task.cpp

```diff
@@ -261,40 +338,56 @@ void LedRingTask::run()
         // This is useful for checking the queue and responding to events without delaying program execution.
         if (xQueueReceive(render_effect_queue_, &effect_settings, 0) == pdTRUE)
         {
-            // Sync reaction to receiving an effect.
-        }
+            if (effect_settings.effect_id == old_effect_settings.effect_id) // PREVENTS LED FLICKERING (only render new effects)
+            {
+                if (effect_settings.effect_id == 7 && effect_settings.effect_brightness == old_effect_settings.effect_brightness)
```

Comment:
Couple of thoughts, for good refactoring opportunity. 

When yo uhave situation like effect_setting.effect_id == 7 in a condition, you really want to go back and change the 7 into a verbose constant. So someone doesn't need to go back and look up what 7 means. 
The if becomes effect_settings.effect_id == EFFECT_LIGHTHOUSE for instance. where EFFECT_LIGHTHOUSE = 7. 

A more complex thought:

Here you have a situation with nested If(s). Even more, there is a switch on effect just after. Always be worried when you see something like this.
It means there  is poor encapsulation somewhere. In this case you can see how the Run method, and the specific effect 7, are sharing the role in determine when the effect is over. You want to figure out a way to move that completley inside the effect itself. And, leave the method run to just execute whatever effect (without having to know the logic of each effect)

Reply:
Just utilized @brushknight original code or whoever wrote it from the start. (Done like u suggest in my other code using enums and switch statements etc.)

But ill refactor into this. 👍